popde/density_estimate.py

```diff
@@ -67,9 +67,6 @@ def __init__(self, data, weights=None, input_transf=None, stdize=False,
 
         self.data = np.asarray(data)
         self.ndim = self.data.shape[1]
-        self.weights = np.atleast_1d(weights).astype(float)
```

Comment:
why don't we want to keep these lines but only perform them if weights is not None? 

I.e. 
```
if weights is not None:
    self.weights = (etc...)
    if len(self.weights) != self.data.shape[0]"
        (etc)
else:
    self.weights = None
```

Reply:
Well, if weights are None than these lines will raise error. These checks are there in next few lines of the code so this is redundant

Reply:
I don't know what you mean Jam. I said perform the checks etc if weights is not None. If weights is a valid option for this class then we should perform the checks here and remove the ones done in a subclass.

Reply:
If you check the next few lines of the code there is this check done. https://github.com/tdent/pop-de/blob/main/popde/density_estimate.py#L82